libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -134,33 +133,18 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
         return navigator!!.updateSensorData(sensorData)
     }
 
-    /**
-     * Gets the status as an offset in time from the last fixed location. This
-     * allows the caller to get predicted statuses in the future along the route if
-     * the device is unable to get fixed locations. Poor reception would be one reason.
-     *
-     * This method uses previous fixes to snap the user's location to the route
-     * and verify that the user is still on the route. This method also determines
-     * if an instruction needs to be called out for the user.
-     *
-     * @param navigatorPredictionMillis millis for navigation status predictions.
-     *
-     * @return the last [TripStatus] as a result of fixed location updates. If the timestamp
-     * is earlier than a previous call, the last status will be returned. The function does not support re-winding time.
-     */
-    override suspend fun getStatus(navigatorPredictionMillis: Long): TripStatus =
+    suspend fun generateTripStatusFrom(navigatorStatus: NavigationStatus): TripStatus =
```

Comment:
We can remove the suspend here right? There is no need to context switch to the `NavigatorDispatcher` because the navigator has already given the SDK the status

Reply:
Actually, this function should be moved to `NavigatorMapper`